### PR TITLE
skill: add crossplane-compositions (Phase 2 — moved + enhanced from realm)

### PR DESCRIPTION
**Pull request by agent-refr**
> **AI-assisted change proposal.** Filed by agent driven by @Cervator via [GDD](https://siliconsaga.github.io/yggdrasil/gdd/).

## Summary

- **Phase 2** of the SiliconSaga skill taxonomy: moves `crossplane-on-k3d` from `realm-siliconsaga` to a root capability skill (renamed `crossplane-compositions`) and enhances with this session's hard-won lessons.
- **Adds**: `crossplane render` for offline validation; the RWO + RollingUpdate → `Recreate` strategy + the SSA in-place migration trap with BOTH fixes (surgical patch vs nuclear delete) and when to use each; test-through-GitOps as the specific case (don't `kubectl apply` over an ArgoCD-managed Composition — the flapping war is real); stale-revision-numbering edge case for `compositionUpdatePolicy: Automatic`; `managedFields` as the diagnostic for "who's writing this?".
- **Preserves** from the realm: operator `watchAllNamespaces` Helm trap, provider→ProviderConfig ordering, `kubectl explain` for CRD verification, security-context overrides, XRD v1 deprecation note.

## Test plan

Per `superpowers:writing-skills` (TDD for documentation):

- [x] **RED** — baseline subagent without skill was remarkably strong (got `crossplane render`, CEL/v1alpha2, Multi-Attach reasoning, SSA-Recreate trap, even `managedFields`). Real gaps: test-through-GitOps as the specific case, both migration fix options (only had surgical), stale-revision pinning, `watchAllNamespaces` operator gotcha, cross-references.
- [x] **GREEN** — skill addresses exactly those gaps + cites authoritative sources + preserves the realm content.
- [x] **VERIFY GREEN** — re-dispatched subagent cited skill sections extensively, gave both migration options with "when to use each" guidance, named all three flapping causes, surfaced a 3rd SSA force-conflicts option and noted the skill recommends against it.
- [x] **REFACTOR** — no new rationalizations to plug.

## Related

- Phase 2 follow-up PR on `realm-siliconsaga` will remove the old `crossplane-on-k3d` skill (now superseded by this).
- Companion Phase 2 skill `argocd-gitops` ships in another PR.
- Phase 1 skills: #78 (`alertmanager-config`), #79 (`kube-prometheus-stack`).

## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive Crossplane Compositions guide covering diagnostic workflows for Synced/Ready issues, stepwise troubleshooting across composite claims and managed resources, and visibility tips for provider-kubernetes.
  * Documented specific failure scenarios (CEL readiness, Deployment strategy migration, CompositionRevision flapping) and common setup pitfalls.
  * Included offline validation guidance, GitOps testing recommendations, an implementation sketch, and references.

### CODE REVIEW

File: .agent/skills/crossplane-compositions/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+| Provider setup order | Install providers → `kubectl wait --for=condition=Healthy providers.pkg.crossplane.io --all --timeout=120s` → THEN apply `ProviderConfig`s | Otherwise `no matches for kind ProviderConfig` (CRDs don't exist yet). |
+| Operator that watches all namespaces | Use the operator's boolean (e.g. `--set watchAllNamespaces=true`), NOT `--set watchNamespace=""` | Helm `--set` silently ignores empty strings → operator only watches its own namespace → Claim stuck Synced=True Ready=False forever. |
+| CRD field shape | `kubectl explain <kind>.spec.<path>` | Don't guess `spec.instances.containers` vs `spec.instances.initContainer`; the API server has the truth. |
+| Testing | kuttl for end-to-end (Claim→Ready) verification | See root `kuttl-testing` skill — `--config kuttl-test.yaml` matters. |
+
+## CEL Readiness Failure Modes
+
+`object.status.<field>` throws `no such key: status` if the operator hasn't written **any** status yet — transient at first, **permanent** if the operator isn't watching the claim's namespace.
```
[the remark this answers is not in the document]
Comment (agent-refr) [reply .agent/skills/crossplane-compositions/SKILL.md:35]: Fixed in a76cc38 — object.X is now consistent throughout (quick-ref row and implementation sketch YAML, failure-mode section was already correct). provider-kubernetes v1alpha2 evaluates the CEL against the observed object with object as the root variable.

**Comment by agent-refr** [comment]
Closed without merging — content ported to its owning component per the revised SiliconSaga skill taxonomy.

The original placement at yggdrasil root followed the design's "highest applicable tier by capability" rule. On review (see [SiliconSaga/realm-siliconsaga#6](https://github.com/SiliconSaga/realm-siliconsaga/pull/6) for the revised design doc), the principle was refined to "highest tier where the knowledge is **genuinely owned**" — and Crossplane Compositions is an opinionated platform choice with competing alternatives, so it belongs with the component that operates it.

This PR's content is ported verbatim (with all review fixes preserved) to **SiliconSaga/nordri#16**. Yggdrasil root stays clean of infra concerns going forward.

Refined design doc: `realms/realm-siliconsaga/docs/plans/2026-05-30-skill-taxonomy-design.md` (revised in SiliconSaga/realm-siliconsaga#6).
